mycore-user2/src/main/java/org/mycore/user2/login/MCRLDAPClient.java

```diff
@@ -258,8 +221,134 @@ private String formatName(String name) {
         return trimmedName.substring(pos + 1) + ", " + trimmedName.substring(0, pos);
     }
 
+    public record ConnectionSettings(
+        String providerUrl,
+        String securityPrincipal,
+        String securityCredentials,
+        Integer readTimeoutMillis) {
+
+        public ConnectionSettings validate() {
+            Objects.requireNonNull(providerUrl, "Provider URL must not be null");
+            return this;
+        }
+
+        public Map<String, String> toLdapSettings() {
+            validate();
+            Map<String, String> ldapEnv = new HashMap<>();
+            ldapEnv.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
+            ldapEnv.put(Context.PROVIDER_URL, providerUrl);
+            ldapEnv.put(Context.SECURITY_AUTHENTICATION, "simple");
+            if (securityPrincipal != null) {
+                ldapEnv.put(Context.SECURITY_PRINCIPAL, securityPrincipal);
+                if (securityCredentials != null) {
+                    ldapEnv.put(Context.SECURITY_CREDENTIALS, securityCredentials);
+                }
+            }
+            ldapEnv.put("com.sun.jndi.ldap.read.timeout",
+                readTimeoutMillis != null ? readTimeoutMillis.toString() : "1000");
+            return ldapEnv;
+        }
+
+    }
+
+    public record Settings(
```

Comment:
For me, the name is a bit confusing because there are already the `ConnectionsSettings`.
May move _Base DN_ as "connection entry" to `ConnectionsSettings` and rename `Settings` in `MappingSettings`.